File: ServerApp/Karuna/Karuna/src/main/java/com/karuna/services/DonorService.java
package com.karuna.services;

import java.util.List;

import com.karuna.dto.ItemDto;
import com.karuna.dto.LoginDto;
import com.karuna.dto.LogoutDto;
import com.karuna.dto.RegisterDonorDto;
import com.karuna.entity.Campaign;
import com.karuna.entity.Donor;
import com.karuna.entity.Item;
import com.karuna.entity.Receiver;

public interface DonorService {

	Donor registerDonor(RegisterDonorDto donor);
	
	List<Campaign> viewCampaign(Boolean status);
	
	Donor updateDonor(RegisterDonorDto updatedDonor);
	
	Donor login(LoginDto loginDto);
	
	Item donate(ItemDto item);
	
	String logout(LogoutDto logoutDto);
	
	String deleteAccount(Long donorId);
	
	List<Receiver> viewReceivers();

	List<Item> viewHistory(Donor donor);
}

File: ClientApp/karuna/src/services/DonorService.js
import axios from 'axios';

const BASE_URL = '/donor';

const DonorService = {
  registerDonor: async (donor) => {
    const response = await axios.post(`${BASE_URL}/register`, donor);
    return response.data;
  },

  viewCampaign: async () => {
    const response = await axios.get(`${BASE_URL}/campaigns`);
    return response.data;
  },

  deleteAccount: async (donorId) => {
    const response = await axios.delete(`${BASE_URL}/${donorId}`);
    return response.data;
  },

  updateDonor: async (donor) => {
    const response = await axios.put(`${BASE_URL}/profile`, donor);
    return response.data;
  },

  donate: async (item) => {
    const response = await axios.post(`${BASE_URL}/donate`, item);
    return response.data;
  },

  login: async (credentials) => {
    const response = await axios.post(`${BASE_URL}/login`, credentials);
    return response.data;
  },

  viewReceivers: async () => {
    const response = await axios.get(`${BASE_URL}/receivers`);
    return response.data;
  },

  viewHistory: async (donorId) => {
    const response = await axios.get(`${BASE_URL}/history/${donorId}`);
    return response.data;
  },

  logout: async (donorId) => {
    const response = await axios.get(`${BASE_URL}/${donorId}`);
    return response.data;
  }
};

export default DonorService;
